metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -334,27 +338,30 @@ def build_corp_user_mce(
         )
         manager_urn = f"urn:li:corpuser:{manager_ldap}" if manager_ldap else None
 
-        return MetadataChangeEvent(
-            proposedSnapshot=CorpUserSnapshotClass(
-                urn=f"urn:li:corpuser:{ldap_user}",
-                aspects=[
-                    CorpUserInfoClass(
-                        active=True,
-                        email=email,
-                        fullName=full_name,
-                        firstName=first_name,
-                        lastName=last_name,
-                        departmentId=department_id,
-                        departmentName=department_name,
-                        displayName=display_name,
-                        countryCode=country_code,
-                        title=title,
-                        managerUrn=manager_urn,
-                    )
-                ],
-            )
+        user_snapshot = CorpUserSnapshotClass(
+            urn=f"urn:li:corpuser:{ldap_user}",
+            aspects=[
+                CorpUserInfoClass(
+                    active=True,
+                    email=email,
+                    fullName=full_name,
+                    firstName=first_name,
+                    lastName=last_name,
+                    departmentId=department_id,
+                    departmentName=department_name,
+                    displayName=display_name,
+                    countryCode=country_code,
+                    title=title,
+                    managerUrn=manager_urn,
+                )
+            ],
         )
 
+        if groups:
```

Comment:
Nice, this is great!

Reply:
Thank you! Sure, I'll update integration test